epicsdbbuilder/const_array.py

```diff
@@ -0,0 +1,92 @@
+from decimal import Decimal
+from .recordbase import quote_string
+from .parameter import Parameter
+
+__all__ = ['ConstArray']
+
+
+class ConstArray:
+    """Constant Link Values. EPICS Base 7.0.4.1 and above.
+
+    Example: PY Source
+    ------------------
+    `r = records.lsi('r', INP=ConstArray(['Plain String not DBLINK']))`
+
+    Example: Generated DB
+    ---------------------
+    `field(INP, ["Plain String not DBLINK"])`
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        self._assert_valid(value)
+        self._value = value
+
+    def _assert_valid(self, value, record=None, fieldname=None):
+        id = 'ConstArray' \
+            if record is None or fieldname is None \
+            else f'ConstArray@{record}.{fieldname} :'
+
+        assert isinstance(value, list) \
+            or isinstance(value, tuple), \
+            f'{id} expects a list or a tuple but [{type(value)}] was supplied.'
+
+        numbers = False
+        strings = False
+        for index, value in enumerate(value):
+            assert value is None \
```

Comment:
`None` is not a valid value, and this occurs in lots of other places.

Reply:
The intention with `None` as a value in array was to skip a value without a need for a filtering the array. Removed.